### PR TITLE
feat(GeolocateField): add input form geolocate field

### PR DESCRIPTION
**Pull request by goldpbear**
Closes: https://github.com/jalMogo/mgmt/issues/189

### CODE REVIEW

File: src/base/static/components/molecules/form-field-types/geolocate-field.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  textAlign: "left",
+  color: "#999",
+});
+
+const GeolocateField = props => {
+  return (
+    <GeolocateButton
+      onClick={() => emitter.emit(constants.TRIGGER_GEOLOCATE_EVENT)}
+    >
+      <FontAwesomeIcon content="\f05b" fontSize="1.5rem" />
```
Comment (jalMogo) [inline src/base/static/components/molecules/form-field-types/geolocate-field.js]: nit: I would prefer to save the unicode as constants within our atoms, and pass in a description of the content through the api (eg: `content="globe"`). It'll make the api easier to work with, and keeps the unicode in one place.

File: src/base/static/libs/maps/mapboxgl-provider.js [hunk 2]
```diff
@@ -1041,7 +1042,12 @@ export default (container, options) => {
     },
 
     getGeolocateControl: () => {
-      return new mapboxgl.GeolocateControl();
+      geolocateControl = new mapboxgl.GeolocateControl();
+      return geolocateControl;
+    },
```
Comment (jalMogo) [inline src/base/static/libs/maps/mapboxgl-provider.js]: Having a getter that sets a global variable and calls on another constructor seems like an anti-pattern.

Perhaps we can update MainMap to replace:
`this._map.addControl(this._map.getGeolocateControl(), "top-left");` 
with a new function, like `addGeolocationControl`:
`this._map.addGeolocationControl("top-left");`

where `addGeolocationControl` is defined in the provider, like:

```javascript
addGeolocationControl = (position) => {
  geolocateControl = new mapboxgl.GeolocateControl()
  map.addControl(geolocateControl, position)
}
```

File: src/base/static/components/molecules/form-field-types/geolocate-field.js (hunk 1 above)
Comment (goldpbear) [reply src/base/static/components/molecules/form-field-types/geolocate-field.js]: Would we need a complete mapping of all the FontAwesome icons in that case? An alternative approach would be to use the classnames that FontAwesome provides, which are more descriptive. 

So we would pass through something like `content="fa-globe"`.

File: src/base/static/libs/maps/mapboxgl-provider.js (hunk 2 above)
Comment (goldpbear) [reply src/base/static/libs/maps/mapboxgl-provider.js]: Ok, that makes sense.

File: src/base/static/components/molecules/form-field-types/geolocate-field.js (hunk 1 above)
Comment (jalMogo) [reply src/base/static/components/molecules/form-field-types/geolocate-field.js]: Yeah, that sounds good

**Pull request merged**